hoot-core/src/main/cpp/hoot/core/ops/GeometryModifierOp.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016, 2017, 2018, 2019 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#ifndef GEOMETRYMODIFIEROP_H
+#define GEOMETRYMODIFIEROP_H
+
+// Hoot
+#include <hoot/core/elements/OsmMap.h>
+#include <hoot/core/ops/OsmMapOperation.h>
+#include <hoot/core/info/OperationStatusInfo.h>
+#include <hoot/core/visitors/geometrymodifiers/GeometryModifierVisitor.h>
+
+// Boost
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+
+namespace hoot
+{
+  class GeometryModifierAction;
+  struct GeometryModifierActionDesc;
+
+  /*
+   * Geometry modifier operation used to run specific processes changing the actual
+   * geometry of elements, e.g. turning ways into polygons, collapsing polygons to points, etc.
+   *
+   * Available processes are implemented in GeometryModifierAction classes which are
+   * registered with the hoot factory.
+   * The GeometryModifierOp constructor instantiates all available actions and reads the json
+   * rules file which defines the filter and arguments for each requested action.
+   *
+   * Multiple actions can be specified in the rules file and the apply() method uses the
+   * GeometryModifierVisitor to go through all map elements to apply each specified action.
+   * Each action is performed on the entire map before it moves on to the next action.
+   */
+  class GeometryModifierOp : public OsmMapOperation, public OperationStatusInfo
```

Comment:
Any class that gets information from the configuration file should inherit from the `Configurable` class and implement that interface.  An example of that is in `hoot/core/ops/NamedOp.h`.  This has the `setConfiguration` function that allows for the configuration to come from other locations other than the default one, which is used all over in testing and other places.

Reply:
Done